code/datums/gamemodes/gangwar.dm

```diff
@@ -1640,18 +2364,99 @@ proc/broadcast_to_all_gangs(var/message)
 		..()
 		return
 
-// /obj/item/chem_grenade/incendiary
-// 	name = "incendiary grenade"
-// 	desc = "A rather volatile grenade that creates a small fire."
-// 	icon = 'icons/obj/items/grenade.dmi'
-// 	icon_state = "incendiary"
-// 	icon_state_armed = "incendiary1"
-// 	stage = 2
-
-// 	New()
-// 		..()
-// 		var/obj/item/reagent_containers/glass/B1 = new(src)
-// 		B1.reagents.add_reagent("infernite", 20)
-// 		beakers += B1
-#undef CASH_DIVISOR
-#undef DEFAULT_MAX_GANG_SIZE
+// GANG TAGS
+
+/obj/decal/gangtag
+	name = "gang tag"
+	desc = "A gang tag, sprayed with nigh-uncleanable heavy metals."
+	density = FALSE
+	anchored = TRUE
+	layer = TAG_LAYER
+	icon = 'icons/obj/decals/graffiti.dmi'
+	icon_state = "gangtag0"
+	var/datum/gang/owners = null
+	var/list/mobs
+	var/heat = 0 // a rough estimation of how regularly this tag has people near it
+	var/image/heatTracker
+	var/active = TRUE
+	/// Deletes all duplicate tags (IE, from the same gang) on this tile
+	proc/delete_same_tags()
+		for(var/obj/decal/gangtag/T in get_turf(src))
+			if(T.owners == src.owners && T != src) qdel(T)
+
+	/// Makes this tag insert, so it no longer provides points.
+	proc/disable()
+		active = FALSE
+		var/datum/client_image_group/imgroup = get_image_group(CLIENT_IMAGE_GROUP_GANGS)
+		imgroup.remove_image(heatTracker)
+		src.heatTracker = null
+		qdel(heatTracker)
+
+	/// Look for & remember players in this gang's sight range
+	proc/find_players()
+		for(var/mob/M in range(GANG_TAG_SIGHT_RANGE, src.loc))
+			if (IN_EUCLIDEAN_RANGE(src,M,GANG_TAG_SIGHT_RANGE))
+				if(M.client && isalive(M))
+					mobs[M] = TRUE //remember mob
+
+	/// Adds heat to this tag based upon how many mobs it's remembered. Then forgets all mobs it's seen and cools down.
+	proc/calculate_heat()
+		heat += length(mobs)
+		heat = round(heat * GANG_TAG_HEAT_DECAY_MUL, 0.01) //slowly decay heat
+		mobs = list()
+		return heat
+
+	proc/apply_score(var/largestHeat)
+		var/mappedHeat
+		if (heat == 0 || largestHeat == 0)
+			mappedHeat = 0
+		else
+			var/pct = heat/largestHeat
+			var/calculatedHeat = log(10,10*pct)*5
+			if (calculatedHeat < 0)
+				mappedHeat = 0
+			else
+				mappedHeat = round(max(0,calculatedHeat))+1
+
+
+		var/score = 0
+		score = ceil(mappedHeat * GANG_TAG_POINTS_PER_HEAT)
+		owners.score_turf += score
+		owners.add_points(score)
+		owners.show_score_maptext(score, get_turf(src))
+		heatTracker.icon_state = "gang_heat_[mappedHeat]"
```

Comment:
This should probably have some comments to explain what all the different heat vars mean